src/main/java/org/cancogenvirusseq/muse/service/DownloadsService.java

```diff
@@ -38,25 +41,45 @@
   final SongScoreClient songScoreClient;
 
   public Flux<DataBuffer> download(DownloadRequest downloadRequest) {
-    val analysisIds = downloadRequest.getAnalysisIds();
-    val studyId = downloadRequest.getStudyId();
-
-    return Flux.fromIterable(analysisIds)
-        .map(UUID::fromString)
-        .flatMap(analysisId -> songScoreClient.getFileSpecFromSong(studyId, analysisId))
+    return Flux.fromIterable(downloadRequest.getAnalysisIds())
+        .flatMap(
+            id ->
+                // get analysis from song and map to fetch result
+                songScoreClient
+                    .getAnalysis(downloadRequest.getStudyId(), id)
+                    .map(analysis -> new DownloadAnalysisFetchResult(id, analysis))
+                    .onErrorResume(
+                        SongScoreClientException.class,
+                        t -> Mono.just(new DownloadAnalysisFetchResult(id, t))))
+        .collectList()
+        .flatMap(
+            results -> {
+              // see if any results have error
+              if (results.stream().anyMatch(this::isFetchResultNotOk)) {
+                return Mono.error(new DownloadAnalysisFetchException(results));
+              }
+              // return if no errors found here
+              return Mono.just(results);
+            })
+        .flatMapMany(Flux::fromIterable)
+        .map(DownloadAnalysisFetchResult::getFileInfo)
+        // file will exist because we already checked for valid analysis
+        .map(Optional::get)
         .flatMap(
             analysisFileResponse -> {
               val objectId = analysisFileResponse.getObjectId();
               return songScoreClient.downloadObject(objectId);
             })
-        .onErrorMap(
-            t -> {
-              log.error(t.getLocalizedMessage(), t);
-              if (t instanceof MuseBaseException) {
-                return t;
-              } else {
-                return new UnknownException();
-              }
-            });
+        .onErrorMap(t -> !(t instanceof MuseBaseException), t -> new UnknownException());
+  }
+
+  // Result isNotOk if analysis is missing, is not published or has no file objects
+  private Boolean isFetchResultNotOk(DownloadAnalysisFetchResult fetchResult) {
+    return fetchResult.getException().isPresent()
+        || fetchResult
+            .getAnalysis()
+            .map(a -> !a.isPublished() || !a.hasFiles())
+            // analysis doesn't exist so return true
+            .orElse(true);
```

Comment:
I realized while playing around with the new client updates that SONG doesn't check if analysis is published.. So instead of getting just the file info... we get the entire analysis (or at least the parts we care about) and check that its valid. Do this for all ids and then continue with download.